### PR TITLE
First official PR

### PR DESCRIPTION
**Pull request by myrfy001**
This is the first PR to review for the datenlords' kernelspace rdma driver.

### CODE REVIEW

File: driver/dtld.c [hunk 1]
```diff
@@ -1,129 +1,234 @@
...
+    struct xdma_dev *xdev = NULL;
+
+    err = dtld_dev_init_xdma(pdev, id, &xdev);
+    if (err)
+	return err;
 
-static int __init dtld_ib_init(void)
+    err = dtld_dev_init_rdma(xdev);
+    if (err)
+	return err;
```
Comment (EAimTY) [inline driver/dtld.c]: The logic to free xdma resources may be missing here

File: driver/dtld_verbs.h [hunk 2]
```diff
@@ -1,33 +1,253 @@
...
+    struct tasklet_struct comp_task;
+    atomic_t num_wq;
+};
+
+struct dtld_sq {
+    int max_wr;
+    int max_sge;
+    int max_inline;
+
+    struct dtld_pool_elem elem;
```
Comment (EAimTY) [inline driver/dtld_verbs.h]: This should be removed

**Review by EAimTY — CHANGES_REQUESTED** [review]

**Comment by EAimTY** [comment]
I am pushing some changes

**Review by EAimTY — APPROVED** [review]

File: driver/dtld.c [hunk 3]
```diff
@@ -1,129 +1,239 @@
...
-	enum ib_mtu mtu;
+    int rv = 0;
+    struct xdma_pci_dev *xpdev = NULL;
+    void *hndl;
+
+    // TODO: why not move xpdev alloc into `xdma_device_open` and make it an atomic one?
+    // and why multi xdev may bind to the same pdev? I think we should refactor it soon.
+    xpdev = xpdev_alloc(pdev);
+    if (!xpdev)
+        return -ENOMEM;
```
Comment (ruiqurm) [inline driver/dtld.c:123]: OK. I will make another commit to move `xpdev_alloc` into `xdma_device_open`.  I wonder why multi `xdev` may bind to the same `pdev`? Can you give an example?

Comment (ruiqurm) [reply driver/dtld.c:123]: I don't have permission to push a commit. So I refactor it in the next PR?

**Review by ruiqurm — APPROVED** [review]

**Review by ruiqurm — APPROVED** [review]

**Review by ruiqurm — APPROVED** [review]

**Review by ruiqurm — APPROVED** [review]

**Review by ruiqurm — APPROVED** [review]

**Review by ruiqurm — APPROVED** [review]
Approve for merging

**Review by ruiqurm — APPROVED** [review]

**Pull request merged**